src/commands/dev-internal.ts

```diff
@@ -0,0 +1,191 @@
+import { relative, resolve } from 'pathe'
+import chokidar from 'chokidar'
+import type { Nuxt } from '@nuxt/schema'
+import { consola } from 'consola'
+import { debounce } from 'perfect-debounce'
+// We are deliberately inlining this code as a backup in case user has `@nuxt/schema<3.7`
+import { writeTypes as writeTypesLegacy } from '@nuxt/kit'
+import { loadKit } from '../utils/kit'
+import { overrideEnv } from '../utils/env'
+import { loadNuxtManifest, writeNuxtManifest } from '../utils/nuxt'
+import { clearBuildDir } from '../utils/fs'
+import { defineCommand } from 'citty'
+import { sharedArgs, legacyRootDirArgs } from './_shared'
+import { RequestListener, Server } from 'http'
+import { toNodeListener } from 'h3'
+import { AddressInfo } from 'net'
+
+export type NuxtDevIPCMessage =
+  | { type: 'nuxt:internal:dev:ready'; port: number }
+  | { type: 'nuxt:internal:dev:loading'; message: string }
+  | { type: 'nuxt:internal:dev:restart' }
+
+const RESTART_RE = /^(nuxt\.config\.(js|ts|mjs|cjs)|\.nuxtignore|\.nuxtrc)$/
+
+export default defineCommand({
+  meta: {
+    name: '_dev',
+    description: 'Run nuxt development server (internal command)',
+  },
+  args: {
+    ...sharedArgs,
+    ...legacyRootDirArgs,
+  },
+  async run(ctx) {
+    if (!process.send) {
+      throw new Error(
+        'This command is only meant to be run in a subprocess. Please use `nuxi dev`',
+      )
+    }
+
+    // Prepare
+    overrideEnv('development')
+    const cwd = resolve(ctx.args.cwd || ctx.args.rootDir || '.')
+
+    // Load nuxt kit
+    const {
+      loadNuxt,
+      buildNuxt,
+      writeTypes = writeTypesLegacy,
+    } = await loadKit(cwd)
+
+    // Handler
+    let serverHandler: undefined | RequestListener
+    const server = new Server((req, res) => {
+      if (!serverHandler) {
+        // This should not be reached!
+        res.statusCode = 503
+        res.end('Nuxt is not ready yet!')
+        return
+      }
+      serverHandler(req, res)
+    })
+
+    const port = await new Promise<number>((resolve) => {
+      server.listen(0, () => {
+        resolve((server.address() as AddressInfo).port)
+      })
+    })
+
+    function sendIPCMessage<T extends NuxtDevIPCMessage>(message: T) {
+      if (process.send) {
+        process.send(message)
+      }
+    }
+
+    let currentNuxt: Nuxt
+    let distWatcher: chokidar.FSWatcher
+    async function _load(reload?: boolean, reason?: string) {
+      const action = reload ? 'Restarting' : 'Starting'
+      const message = `${reason ? reason + '. ' : ''}${action} nuxt...`
+      serverHandler = undefined
+      sendIPCMessage({
+        type: 'nuxt:internal:dev:loading',
+        message,
+      })
+      if (reload) {
+        consola.info(message)
+      }
+
+      if (currentNuxt) {
+        await currentNuxt.close()
```

Comment:
this might not be an issue with this PR, but got this behaviour:

```
❯ pnpm nuxi dev playground
Nuxt 3.6.5 with Nitro 2.5.2                                                                                                    09:50:06
[get-port] Unable to find an available port (tried 3000). Using alternative port: 3004                                         09:50:06
                                                                                                                               09:50:06
  ➜ Local:    http://localhost:3004/
  > Network: use --host to expose

ℹ nuxt.config.ts updated. Restarting nuxt...                                                                                  09:50:10

 ERROR  Cannot restart nuxt:  Cannot read properties of undefined (reading 'close')                                            09:50:10

  at node_modules/.pnpm/nitropack@2.5.2/node_modules/nitropack/dist/shared/nitro.7a525996.mjs:2416:19
  at node_modules/.pnpm/hookable@5.5.3/node_modules/hookable/dist/index.cjs:50:66
  at node_modules/.pnpm/hookable@5.5.3/node_modules/hookable/dist/index.cjs:50:56
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at async _load (node_modules/.pnpm/file+..+nuxt+cli+nuxi-ng-0.2.0.tgz/node_modules/nuxi-ng/dist/chunks/dev-internal.mjs:162:9)
  at async load (node_modules/.pnpm/file+..+nuxt+cli+nuxi-ng-0.2.0.tgz/node_modules/nuxi-ng/dist/chunks/dev-internal.mjs:218:9)
  at async _applyPromised (node_modules/.pnpm/file+..+nuxt+cli+nuxi-ng-0.2.0.tgz/node_modules/nuxi-ng/dist/chunks/dev-internal.mjs:103:10)
```

Reply:
not now reproducible - I wonder if resolved in nitro 2.6...